cmd/skynet/main.go

```diff
@@ -112,15 +139,25 @@ on top of Sia.`,
 	}
 
 	// Add Skynet Commands
-	rootCmd.AddCommand(skynetBlacklistCmd, skynetConvertCmd, skynetDownloadCmd, skynetLsCmd, skynetPinCmd, skynetUnpinCmd, skynetUploadCmd)
+	rootCmd.AddCommand(skynetBlacklistCmd, skynetConvertCmd, skynetDownloadCmd, skynetLsCmd, skynetPinCmd, skynetSkykeyCmd, skynetUnpinCmd, skynetUploadCmd)
+	skynetSkykeyCmd.AddCommand(skynetSkykeyAddCmd, skynetSkykeyCreateCmd, skynetSkykeyGetCmd)
+	skynetSkykeyGetCmd.AddCommand(skynetSkykeyGetIDCmd, skynetSkykeyGetNameCmd, skynetSkykeyGetSkykeysCmd)
 
-	// Add Flags
+	// Add flags.
 	rootCmd.Flags().BoolVarP(&generateDocs, "", "d", false, "Generate the docs for skynet")
 	rootCmd.PersistentFlags().StringVar(&skynetPortal, "portal", "", "Use a Skynet portal other than the default")
-	skynetUploadCmd.Flags().StringVar(&portalUploadPath, "upload-path", "", "Relative URL path of the upload endpoint")
+	rootCmd.PersistentFlags().StringVar(&endpointPath, "endpoint-path", "", "Relative URL path of the endpoint to use")
+	rootCmd.PersistentFlags().StringVar(&customUserAgent, "custom-user-agent", "", "Custom user agent to use")
+	// Upload flags.
 	skynetUploadCmd.Flags().StringVar(&portalFileFieldName, "file-field-name", "", "Defines the fieldName for the files on the portal")
 	skynetUploadCmd.Flags().StringVar(&portalDirectoryFileFieldName, "directory-field-name", "", "Defines the fieldName for the directory files on the portal")
 	skynetUploadCmd.Flags().StringVar(&customFilename, "filename", "", "Custom filename for the uploaded file")
+	skynetUploadCmd.Flags().StringVar(&customDirname, "dirname", "", "Custom dirname for the uploaded directory")
+	skynetUploadCmd.Flags().StringVar(&uploadSkykeyName, "skykey-name", "", "Name of the skykey on the portal used to encrypt the upload")
```

Comment:
I think we could probably just make these a common PersistentFlag for the root command. There might be more uses of skykeys in the future and then we don't need to keep adding the same flag for all the sub levels.

Reply:
What uses could there be apart from upload and download? I think they shouldn't be root flags because they're not valid for most commands and it would be confusing for users.

Reply:
I don't know yet, but I'm just imagining that there will be more uses. If we think of skykeys as access tokens then they could be used to restrict a lot of things. Fine to leave it as a future consideration. Was just a thought. Not a big deal to update in the future if it does become highly used.